gwcs/api.py

```diff
@@ -27,13 +27,17 @@ def pixel_n_dim(self):
         """
         The number of axes in the pixel coordinate system.
         """
+        if self.input_frame is None:
```

Comment:
Is this the same as if the input frame is of type str?

Reply:
Yes. frames are object attributes or `None`. Given how much functionality depends on the information stored in frames, strings should not be used in general. So `input_frame`, `output_frame` and all frames in `WCS.available_frames` return `CoordinateFrame` objects or `None`.